SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -599,6 +604,82 @@ FString GetComponentSetOutputPathBySchemaType(const FString& BasePath, ESchemaCo
 	return FPaths::Combine(*BasePath, FPaths::Combine(TEXT("ComponentSets"), *FileName));
 }
 
+bool RegexMatch(const FString& Str, const FString& Pattern, TArray<FString>& Result)
+{
+	FRegexPattern MatherPatter(Pattern);
+	FRegexMatcher Matcher(MatherPatter, Str);
+
+	while (Matcher.FindNext())
+	{
+		Result.Add(Matcher.GetCaptureGroup(1));
+	}
+
+	return Result.Num() == 0 ? false : true;
+}
+
+void ImportHandWrittenProjectSchema()
+{
+	const FString HandWrittenSchemaPath = GetDefault<USpatialGDKEditorSettings>()->GetProjectDefinedSchemaFolder();
+	const FString SchemaOutputPath = GetDefault<USpatialGDKEditorSettings>()->GetSchemaCustomFolder();
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+
+	if (!PlatformFile.CopyDirectoryTree(*SchemaOutputPath, *HandWrittenSchemaPath, true))
+	{
+		UE_LOG(LogSpatialGDKSchemaGenerator, Error,
+			   TEXT("Could not Copy Hand Written Project Schema '%s'! Please make sure the directory and the files inside are writeable."),
+			   *SchemaOutputPath);
+	}
+	else
+	{
+		TArray<FString> CustomSchemaFiles;
+		const TCHAR* FileExtension = TEXT(".schema");
+		ServerAuthorityWellKnownComponents_Custom.Empty();
+		ServerAuthorityWellKnownSchemaImports_Custom.Empty();
+
+		PlatformFile.FindFilesRecursively(CustomSchemaFiles, *SchemaOutputPath, FileExtension);
+		for (FString SchemaFile : CustomSchemaFiles)
+		{
+			FString SchemaImportsAddition;
+			FString PathToSchemaAdd;
+			SchemaFile.Split(FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("schema/")), &PathToSchemaAdd,
+							 &SchemaImportsAddition);
+
+			// add the custom schema path to the server's well known imports list
+			TArray<FString>(ServerAuthorityWellKnownSchemaImports_Custom).Add(SchemaImportsAddition);
+			FString SchemaFileContentString;
+
+			FFileHelper::LoadFileToString(SchemaFileContentString, *SchemaFile);
+			FString PackageNameBase = "";
+			TArray<FString> ResultPackage;
```

Comment:
```suggestion
TArray<FString> MatchedPackagePath;
```

Once we look for only one match, this would also replace `PackageNameBase` above.